include/openmc/plot.h

```diff
@@ -319,31 +378,154 @@ class ProjectionPlot : public PlottableInterface {
     {}
   };
 
-  // Max intersections before we assume ray tracing is caught in an infinite
-  // loop:
-  static const int MAX_INTERSECTIONS = 1000000;
-
-  std::array<int, 2> pixels_;              // pixel dimension of resulting image
-  double horizontal_field_of_view_ {70.0}; // horiz. f.o.v. in degrees
-  Position camera_position_;               // where camera is
-  Position look_at_;             // point camera is centered looking at
-  Direction up_ {0.0, 0.0, 1.0}; // which way is up
-
   // which color IDs should be wireframed. If empty, all cells are wireframed.
   vector<int> wireframe_ids_;
 
-  /* The horizontal thickness, if using an orthographic projection.
-   * If set to zero, we assume using a perspective projection.
-   */
-  double orthographic_width_ {0.0};
-
   // Thickness of the wireframe lines. Can set to zero for no wireframe.
   int wireframe_thickness_ {1};
 
   RGBColor wireframe_color_ {BLACK}; // wireframe color
   vector<double> xs_; // macro cross section values for cell volume rendering
 };
 
+/*
+ * Plots a geometry with single-scattered Phong lighting
+ * plus a diffuse lighting contribution. Makes for a visually
+ * appealing 3D view of a geometry
+ */
+class PhongPlot : public RayTracePlot {
+  friend class PhongRay;
+
+public:
+  PhongPlot(pugi::xml_node plot);
+
+  virtual void create_output() const;
+  virtual void print_info() const;
+
+  /* All materials are completely transparent unless explicitly listed
+   * to be otherwise
+   */
+  bool is_id_opaque(int id) const;
+
+private:
+  void set_opaque_ids(pugi::xml_node node);
+  void set_light_position(pugi::xml_node node);
+  void set_diffuse_fraction(pugi::xml_node node);
+
+  // TODO
+  // Right now we use a sorted vector and do binary search,
```

Comment:
I'm actually just going to use `std::set` instead if you think it's important enough to merit an issue! I was hoping to avoid having to implement a set class if we ever wanted to run this on GPU, but honestly, making archaic design decisions and having everything avoid modern C++ just with this in mind doesn't really make sense. So, yes, it is now just using std::set.

Reply:
Agreed! That would be my preference but I didn't want to ask more of you at this point given that the feature is pretty complete. Thanks for making this update!